benchmark.py

```
@@ -0,0 +1,826 @@
+            return exact_subset
+        else:
+            # Generate completely random hashes (should not match)
+            # These should not have significant overlap with any inserted fingerprint
+            hashes = []
+            for _ in range(self.rng.randint(self.config.min_hashes_per_fingerprint, self.config.max_hashes_per_fingerprint)):
+                hashes.append(self.rng.randint(0, self.config.max_hash_value))
+            return hashes
+
+
+async def generate_fingerprints(num_docs: int, generator: FingerprintGenerator) -> AsyncGenerator[dict[str, Any], None]:
+    for i in range(1, num_docs + 1):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)